x-pack/plugins/ml/public/application/memory_usage/memory_tree_map/tree_map.tsx

```diff
@@ -65,13 +64,7 @@ export const JobMemoryTreeMap: FC<Props> = ({ node, type, height }) => {
   } = useMlKibana();
   const isDarkTheme = useIsDarkTheme(themeService);
 
-  const { theme, baseTheme } = useMemo(
-    () =>
-      isDarkTheme
-        ? { theme: EUI_CHARTS_THEME_DARK, baseTheme: DARK_THEME }
-        : { theme: EUI_CHARTS_THEME_LIGHT, baseTheme: LIGHT_THEME },
-    [isDarkTheme]
-  );
+  const baseTheme = useMemo(() => (isDarkTheme ? DARK_THEME : LIGHT_THEME), [isDarkTheme]);
```

Comment:
It's not this specific chart, but there is an extra border now showing on this chart in ML (Overview page and Memory usage page), which on `main` only shows on hover:

This PR:

![image (28)](https://github.com/elastic/kibana/assets/7405507/70d5341e-e1dd-41b7-87d8-9d0060eabdd9)


`main`:

<img width="680" alt="image" src="https://github.com/elastic/kibana/assets/7405507/ec438730-c693-4d7b-ab84-a9ceb5cec18c">

Reply:
In this case, this chart was not wired up to any theme so it picked up the newest chart theme which is why the chart margins went away. Also since this chart is not using the correct light/dark theme, you see the very bright hover colors.

https://github.com/elastic/kibana/blob/eac36e8024db3f3048d74dd9e3f1eff6196177dd/x-pack/plugins/ml/public/application/memory_usage/nodes_overview/memory_preview_chart.tsx#L119-L123

To fix the margins I just added the `LEGACY_LIGHT_THEME.chartMargins` to the `Settings.theme`. I also hooked this chart up the the dark-mode specific theme to improve the dark mode theming. See changes in b6ed57602d40b151a6ba7acfe9982fcbf9b1f19c.

![Screen Recording 2023-12-01 at 10 45 32 AM](https://github.com/elastic/kibana/assets/19007109/f08b67d8-acfd-4a27-bd90-673756b6d884)


Thanks for pointing this out, I think this case is rare but I'll check throughout ML and kibana if there are other cases and fix them as well.

Reply:
Looks like there's a few cases of this! https://github.com/elastic/kibana/issues/172385